yagb-core/src/emulator/emulator.ts

```diff
@@ -30,15 +30,15 @@ export interface BusTrap {
 }
 
 export class Emulator {
-    constructor(cartridgeImage: Uint8Array, printCb: (message: string) => void, savedRam?: Uint8Array) {
+    constructor(cgbSupport: boolean, cartridgeImage: Uint8Array, printCb: (message: string) => void, savedRam?: Uint8Array) {
         this.system = new System(printCb);
 
         const cartridge = createCartridge(cartridgeImage, this.system);
         if (!cartridge) {
             throw new Error('bad cartridge image');
         }
 
-        this.mode = cartridge.cgbSupportLevel === CgbSupportLevel.none ? Mode.dmg : Mode.cgb;
+        this.mode = cgbSupport && cartridge.cgbSupportLevel !== CgbSupportLevel.none ? Mode.cgb : Mode.dmg;
```

Comment:
I think the logic here should rather look like:

* No CGB support in cart? Always DMG (or later DMG or DMG with CGB palettes, depending on the flag)
*  CGB compatible? DMG or CGB according to flag
* CGB only? Always CGB